pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -162,32 +164,75 @@ func (r *ReconcileKogitoApp) Reconcile(request reconcile.Request) (reconcile.Res
 		},
 	})
 	if err != nil {
-		return reconcile.Result{}, err
+		updateResourceResult.Err = err
+		updateResourceResult.ErrorReason = v1alpha1.ParseCRRequestFailedReason
 	}
 
-	if kogitoResources.BuildConfigS2IStatus.IsNew {
-		log.Infof("Buildconfigs are created, triggering build %s", kogitoResources.BuildConfigS2I.Name)
-		if _, err := openshift.BuildConfigC(r.client).TriggerBuild(kogitoResources.BuildConfigS2I, instance.Name); err != nil {
-			return reconcile.Result{}, err
+	deployedRes, err := kogitores.GetDeployedResources(instance, r.client.ControlCli)
+	if err != nil {
+		updateResourceResult.Err = err
+		updateResourceResult.ErrorReason = v1alpha1.RetrieveDeployedResourceFailedReason
+	}
+
+	requestedRes := compare.NewMapBuilder().Add(k8sRes...).ResourceMap()
+
+	comparator := kogitores.GetComparator()
+	deltas := comparator.Compare(deployedRes, requestedRes)
+
+	writer := write.New(r.client.ControlCli).WithOwnerController(instance, r.scheme)
+
+	var hasUpdates bool
+	for resourceType, delta := range deltas {
+		if !delta.HasChanges() {
+			continue
 		}
+		log.Infof("Will create %d, update %d, and delete %d instances of %v", len(delta.Added), len(delta.Updated), len(delta.Removed), resourceType)
+		added, err := writer.AddResources(delta.Added)
+		if err != nil {
+			updateResourceResult.Err = err
+			updateResourceResult.ErrorReason = v1alpha1.CreateResourceFailedReason
+		}
+		updated, err := writer.UpdateResources(deployedRes[resourceType], delta.Updated)
+		if err != nil {
+			updateResourceResult.Err = err
+			updateResourceResult.ErrorReason = v1alpha1.UpdateResourceFailedReason
+		}
+		removed, err := writer.RemoveResources(delta.Removed)
+		if err != nil {
+			updateResourceResult.Err = err
+			updateResourceResult.ErrorReason = v1alpha1.RemoveResourceFailedReason
+		}
+		hasUpdates = hasUpdates || added || updated || removed
 	}
 
-	var resourcesUpdateResult *kogitores.UpdateResourcesResult
-	log.Infof("Handling changes in Kogito App '%s'", instance.Name)
-	resourcesUpdateResult = kogitores.ManageResources(instance, kogitoResources, r.client)
+	bcDelta := deltas[reflect.TypeOf(obuildv1.BuildConfig{})]
+	if bcDelta.HasChanges() {
+		var bcs []*obuildv1.BuildConfig
+		for _, bc := range bcDelta.Added {
+			bcs = append(bcs, bc.(*obuildv1.BuildConfig))
+		}
+		for _, bc := range bcDelta.Updated {
+			bcs = append(bcs, bc.(*obuildv1.BuildConfig))
+		}
+
+		if err = r.triggerBuilds(instance, bcs...); err != nil {
+			updateResourceResult.Err = err
+			updateResourceResult.ErrorReason = v1alpha1.TriggerBuildFailedReason
+		}
+	}
 
 	log.Infof("Handling Status updates on '%s'", instance.Name)
-	statusUpdateResult := status.ManageStatus(instance, kogitoResources, r.client, resourcesUpdateResult, r.cache, request.NamespacedName)
+	statusUpdateResult := status.ManageStatus(instance, kogitoResources, r.client, updateResourceResult, r.cache, request.NamespacedName)
```

Comment:
I guess we could have all the status management in one place?

Reply:
Thanks @ricardozanini . The status management is handled in one place by `status.ManageStatus`. I have changed the [code](https://github.com/kiegroup/kogito-cloud-operator/blob/fd37c95a0c0d6ba5a55f92c6e57b2e39c1bd1e8b/pkg/controller/kogitoapp/kogitoapp_controller.go#L214) to assign the `hasUpdates` value to `updateResourceResult` which is an argument of `status.ManageStatus`.